### ISSUE TITLE
Orchestrations Stuck in Pending after Rewinding

### ISSUE THREAD

**Issue by bijeebuss** [Azure/azure-functions-durable-extension#1557]
### Description
> After Rewinding a Failed Orchestration the function gets stuck in Pending Status

### Expected behavior
> After Rewinding the Orchestration it should restart and retry the failed activity function

### Actual behavior
> After Rewinding a Failed Orchestration the function get stuck in Pending Status

### Known workarounds
> Delete the orchestration and create a new one with the same parameters

### App Details

-     <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.DurableTask" Version="2.2.2" />

-     <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.9" />

-  C#

### Azure

- **2020-11-10T21:12:34.200Z**
- **West US 2**
- Example Pending Orchestration ID: **8e43f4adf786480cb36b6d608e990243**:
- Example Failed Instance that will likely also get stuck if rewound: **a4bb2e16b9a64848b0887f792a617e44**

Please email me privately if you need the function app name at mwelnick@digitalairstrike.com

**Comment by cgillum** [comment]
@bijeebuss Thanks for this info. I found your app in our telemetry and I see that you issued the rewind command at **2020-11-10 21:12:31.377**. I also see that the orchestration resumed successfully after the rewind and performed several actions:

* Called EnqueueTrackingItem (2020-11-10 21:12:32.209)
* Received event TrackingQueueCompleted (2020-11-10 21:12:32.887)
* Called ProcessTrackingItem (2020-11-10 21:12:33.294)
* Started OrchestrateTrackingNotification (2020-11-10 21:12:33.712)
* Received RaiseEvent:NotificationCompleted (2020-11-10 21:12:34.052)
* Called SaveNotificationResult (2020-11-10 21:12:34.053)
* Waiting for the next external event (2020-11-10 21:12:34.188)

I can't tell from our telemetry what the actual runtime status is. If it says "Pending" like you mentioned, then this might be a bug where we didn't correctly update the runtime status to "Running" even though it started running successfully.

Does this analysis make sense based on what you're seeing? If so, I think you can repair this by manually updating the **CommunicationApiInstances** table in Azure Storage to update the runtime status field to `Running` for the entity with PartitionKey = '8e43f4adf786480cb36b6d608e990243'.

If you can confirm, then we can look into trying to reproduce and fixing the status update issue with the Rewind API.

> MSFT Internal: [Kusto link](https://wawswus.kusto.windows.net/wawsprod?query=H4sIAAAAAAAEAG2RX2vCMBTF3wf7DhefLFRJtevqwAeZHSvYTWbZe5pea2dNXXPrH9iHX2K1ghuEhqS%2fc%2b7JvdO64kmBL7UUlJdSBTuUpO5%2bYL%2fCCiEOo2ART6I5JEh7RAndlBNSvsHugA1Yz3F6DoOB88SYXhb0%2b%2bCsrFYfSkVcCgxTGI%2bh46M7XLo8XT76nusz)

**Comment by bijeebuss** [comment]
You are right, it did actually process the Rewind but I can see the status in azure storage explorer as 'Pending'

**Comment by cgillum** [comment]
@bijeebuss thanks for confirming. I will leave this issue open to track fixing the bug that is causing the status to not be updated to "Running". In the meantime, the way to repair this instance is to update the record in Azure Storage directly as I described previously.

**Comment by andrewdmoreno** [comment]
I had a similar issue just now (orchestration stuck after rewind). I performed the recommendation @cgillum recommended previously to update storage explorer value to 'Running' and can confirm this did work for me.
